rllib/agents/sac/sac_tf_policy.py

```diff
@@ -40,45 +39,56 @@ def build_sac_model(policy, obs_space, action_space, config):
             "Consider reshaping this into a single dimension, "
             "using a Tuple action space, or the multi-agent API.")
 
+    # 2 cases:
+    # 1) with separate state-preprocessor (before obs+action concat).
+    # 2) no separate state-preprocessor: concat obs+actions right away.
     if config["use_state_preprocessor"]:
-        default_model = None  # catalog decides
-        num_outputs = 256  # arbitrary
-        config["model"]["no_final_linear"] = True
+        num_outputs = 256  # Flatten last Conv2D to this many nodes.
     else:
-        default_model = NoopModel
-        num_outputs = int(np.product(obs_space.shape))
+        config["model"]["fcnet_hiddens"] = []
+        num_outputs = 0
 
+    # Force-ignore any additionally provided hidden layer sizes.
+    # Everything should be configured using SAC's "Q_model" and "policy_model"
+    # settings.
     policy.model = ModelCatalog.get_model_v2(
-        obs_space,
-        action_space,
-        num_outputs,
-        config["model"],
-        framework="tf",
-        model_interface=SACModel,
-        default_model=default_model,
+        obs_space=obs_space,
+        action_space=action_space,
+        num_outputs=num_outputs,
+        model_config=config["model"],
+        framework="torch" if config["use_pytorch"] else "tf",
+        model_interface=SACTorchModel if config["use_pytorch"] else SACTFModel,
         name="sac_model",
-        actor_hidden_activation=config["policy_model"]["hidden_activation"],
-        actor_hiddens=config["policy_model"]["hidden_layer_sizes"],
-        critic_hidden_activation=config["Q_model"]["hidden_activation"],
-        critic_hiddens=config["Q_model"]["hidden_layer_sizes"],
+        actor_hidden_activation=config["policy_model"]["fcnet_activation"],
+        actor_hiddens=config["policy_model"]["fcnet_hiddens"],
+        critic_hidden_activation=config["Q_model"]["fcnet_activation"],
+        critic_hiddens=config["Q_model"]["fcnet_hiddens"],
         twin_q=config["twin_q"],
-        initial_alpha=config["initial_alpha"])
+        initial_alpha=config["initial_alpha"],
+        target_entropy=config["target_entropy"])
+
+    if not config["use_pytorch"]:
+        policy.model.action_model.summary()
+        policy.model.q_net.summary()
+    else:
+        print(policy.model.action_model)
+        print(policy.model.q_net)
```

Comment:
Stray prints here?

Reply:
Actually, we shouldn't print this here, since it will be printed multiple times for each worker, and disrespects the log level. The right place is probably  `_debug_vars` in tf_policy, which throttles it with log_once() and also and logs it at INFO level.

Reply:
I'll remove this entirely. It was only for debugging anyways.